### PR TITLE
fixed bug where search box wasn't getting cleared

### PR DESCRIPTION
**Pull request by yusra-haider**

### CODE REVIEW

**Comment by yusra-haider** [comment]
corresponding ticket: https://app.zenhub.com/workspaces/dcp-ingest-product-development-5f71ca62a3cb47326bdc1b5c/issues/ebi-ait/dcp-ingest-central/397

File: client/src/app/all-projects/all-projects.component.html [hunk 1]
```diff
@@ -10,8 +10,8 @@ <h1 class="vf-text vf-text-heading--2">All Projects</h1>
 
   <mat-form-field class="search-bar">
     <mat-icon matPrefix>search</mat-icon>
-    <input matInput placeholder="Search Projects" (keydown.enter)="onSearch($event.target.value)">
-    <button mat-button matSuffix mat-icon-button aria-label="Clear" (click)="onSearch('')">
+    <input matInput placeholder="Search Projects" [(ngModel)]="searchText" (keydown.enter)="onSearch(searchText)">
+    <button mat-button *ngIf="searchText" matSuffix mat-icon-button aria-label="Clear" (click)="searchText=''; onSearch(searchText)">
```
Comment (amnonkhen) [inline client/src/app/all-projects/all-projects.component.html:14]: It would be a good exercise to write a test for this.

**Review by amnonkhen — CHANGES_REQUESTED** [review]

**Review by amnonkhen — DISMISSED** [review]
well done

File: client/src/app/all-projects/all-projects.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [ AllProjectsComponent],
+      providers: [ { provide: IngestService, useClass: MockIngestService } ]
+    })
+      .compileComponents();
+  }));
+
+  beforeEach(() => {
```
Comment (amnonkhen) [inline client/src/app/all-projects/all-projects.component.spec.ts:18]: Why 2 `beforeEach` parts?

**Review by amnonkhen — CHANGES_REQUESTED** [review]

File: client/src/app/all-projects/all-projects.component.spec.ts (hunk 2 above)
Comment (jacobwindsor) [reply client/src/app/all-projects/all-projects.component.spec.ts:18]: One is async. This is the standard boilerplate

File: client/src/app/all-projects/all-projects.component.spec.ts [hunk 3]
```diff
@@ -25,10 +25,23 @@ describe('AllProjectsComponent', () => {
...
+  });
+
   it('should call onClearSearch', fakeAsync(() => {
     spyOn(component, 'onClearSearch');
-    const button = fixture.nativeElement.querySelector('button');
-    button.click();
+    component.searchText = 'sample search text';
+    fixture.detectChanges();
+    const clearButton = fixture.nativeElement.querySelector('button');
+    clearButton.click();
```
Comment (jacobwindsor) [inline client/src/app/all-projects/all-projects.component.spec.ts:44]: This should check that searchText is empty and that the value of the input field is empty too

**Review by jacobwindsor — CHANGES_REQUESTED** [review]

File: client/src/app/all-projects/all-projects.component.spec.ts (hunk 3 above)
Comment (jacobwindsor) [reply client/src/app/all-projects/all-projects.component.spec.ts:44]: Oh my bad! I didn't see the test below.

File: client/src/app/all-projects/all-projects.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    tick();
+    expect(component.onClearSearch).toHaveBeenCalled();
+  }));
+
+  it('#onClearSearch() should clear search text', () => {
+    component.onClearSearch();
+    expect(component.searchText).toBe('');
```
Comment (jacobwindsor) [inline client/src/app/all-projects/all-projects.component.spec.ts:51]: Just check that the actual value of the input field is empty too

**Review by jacobwindsor — CHANGES_REQUESTED** [review]

File: client/src/app/all-projects/all-projects.component.spec.ts (hunk 4 above)
Comment (yusra-haider) [reply client/src/app/all-projects/all-projects.component.spec.ts:51]: `searchText` is the ngmodel in this case, so I'm not quite sure what else to check?

Comment (jacobwindsor) [reply client/src/app/all-projects/all-projects.component.spec.ts:51]: Maybe it's unnecessary but something like `expect(fixture.nativeElement.querySelector('input').value).toBe('')`. It's a bit belt and braces

**Review by ke4 — DISMISSED** [review]
Looks fine by me

**Review by aaclan-ebi — DISMISSED** [review]

**Pull request merged**